allensdk/api/cloud_cache/manifest.py

```diff
@@ -0,0 +1,189 @@
+import json
+import pathlib
+import copy
+from typing import Union
+from allensdk.api.cloud_cache.utils import relative_path_from_uri  # noqa: E501
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes  # noqa: E501
+
+
+class Manifest(object):
+    """
+    A class for loading and manipulating the online manifest.json associated
+    with a dataset release
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        The path to the directory where local copies of files will be stored
+    """
+
+    def __init__(self, cache_dir: Union[str, pathlib.Path]):
+        if isinstance(cache_dir, str):
+            self._cache_dir = pathlib.Path(cache_dir).resolve()
+        elif isinstance(cache_dir, pathlib.Path):
+            self._cache_dir = cache_dir.resolve()
+        else:
+            raise ValueError("cache_dir must be either a str "
+                             "or a pathlib.Path; "
+                             f"got {type(cache_dir)}")
+
+        self._data = None
```

Comment:
Could you please type these so it's easier to know what the expected data types are?

Reply:
Done